Unable to run testcases with testng 7.0.0-beta1 in eclipse / maven test

I am using latest testng 7.0.0-beta1 version to show factory parameters in custom report. Below is the dependency in pom.xml

		<dependency>
			<groupId>org.testng</groupId>
			<artifactId>testng</artifactId>
			<version>7.0.0-beta1</version>
			<scope>test</scope>
		</dependency>

Eclipse plugin details 

- TestNG	7.0.0.201809110942	org.testng.eclipse.feature.group	Cedric Beust
-   TestNG M2E (Maven) Integration (Optional)	7.0.0.201809110942	org.testng.eclipse.maven.feature.feature.group	TestNG Team
-   TestNG P2 Feature	7.0.0.b201809110940	org.testng.p2.feature.feature.group	TestNG Team


Test case execution is failing in both eclipse and command line. 
In eclipse when running from a testng xml I am getting this exception:

[RemoteTestNG] failed to get TestNG version from class: java.lang.IllegalArgumentException: invalid version "7.0.0-": non-numeric "0-"
[RemoteTestNG] now trying to parse the version from pom.properties
[RemoteTestNG] now trying to parse the version from MANIFEST.MF
Exception in thread "main" java.lang.RuntimeException: Can't recognize the TestNG version on classpath. Please make sure that there's a supported TestNG version (aka. >= 6.0.0) on your project.
	at org.testng.remote.RemoteTestNG.getTestNGVersion(RemoteTestNG.java:121)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:49)


When I am running using mvn test from command line I am getting the below exception:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project PaymentPageUITests: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test failed: There was an error in the forked process
[ERROR] org.testng.TestNGException: 
[ERROR] 
[ERROR] An error occurred while instantiating class com.nttdhktest.DummyTests. Check to make sure it can be instantiated
[ERROR] 	at org.testng.internal.ClassHelper.createInstance1(ClassHelper.java:388)
[ERROR] 	at org.testng.internal.ClassHelper.createInstance(ClassHelper.java:336)
[ERROR] 	at org.testng.internal.ClassImpl.getDefaultInstance(ClassImpl.java:110)
[ERROR] 	at org.testng.internal.ClassImpl.getInstances(ClassImpl.java:188)
[ERROR] 	at org.testng.TestClass.getInstances(TestClass.java:94)
[ERROR] 	at org.testng.TestClass.initTestClassesAndInstances(TestClass.java:79)
[ERROR] 	at org.testng.TestClass.init(TestClass.java:71)
[ERROR] 	at org.testng.TestClass.<init>(TestClass.java:36)
[ERROR] 	at org.testng.TestRunner.initMethods(TestRunner.java:461)
[ERROR] 	at org.testng.TestRunner.init(TestRunner.java:340)
[ERROR] 	at org.testng.TestRunner.init(TestRunner.java:293)
[ERROR] 	at org.testng.TestRunner.<init>(TestRunner.java:178)
[ERROR] 	at org.testng.SuiteRunner$DefaultTestRunnerFactory.newTestRunner(SuiteRunner.java:603)
[ERROR] 	at org.testng.SuiteRunner.init(SuiteRunner.java:194)
[ERROR] 	at org.testng.SuiteRunner.<init>(SuiteRunner.java:125)
[ERROR] 	at org.testng.TestNG.createSuiteRunner(TestNG.java:1230)
[ERROR] 	at org.testng.TestNG.createSuiteRunners(TestNG.java:1209)
[ERROR] 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1058)
[ERROR] 	at org.testng.TestNG.runSuites(TestNG.java:997)
[ERROR] 	at org.testng.TestNG.run(TestNG.java:965)
[ERROR] 	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:281)
[ERROR] 	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
[ERROR] 	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:121)
[ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
[ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
[ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
[ERROR] 

Please help!

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -1,4 +1,5 @@
 Current
+Fixed: GITHUB-1924: Testclass instantiation fails when both no-arg constructor and factory method present (Krishnan Mahadevan)
 Fixed: GITHUB-1935: Wrong text for assertEquals (Krishnan Mahadevan)
 Fixed: GITHUB-1931: [NPE] Reporter org.testng.reporters.jq.Main failed (Krishnan Mahadevan, Oleg Shaburov)
 New: Remove raw type warnings in ConversionUtils

--- a/src/main/java/org/testng/internal/ClassImpl.java
+++ b/src/main/java/org/testng/internal/ClassImpl.java
@@ -184,14 +184,14 @@ public class ClassImpl implements IClass {
                   create)
             };
       }
+    }
+    if (m_instances.size() > 0) {
+      result = m_instances.toArray(new Object[0]);
     } else {
       Object defaultInstance = getDefaultInstance(create);
       if (defaultInstance != null) {
         result = new Object[] {defaultInstance};
       }
-    }
-    if (m_instances.size() > 0) {
-      result = m_instances.toArray(new Object[0]);
     }
 
     int m_instanceCount = m_instances.size();

--- a/src/test/java/test/factory/issue1924/IssueTest.java
+++ b/src/test/java/test/factory/issue1924/IssueTest.java
@@ -1,0 +1,23 @@
+package test.factory.issue1924;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import test.SimpleBaseTest;
+
+public class IssueTest extends SimpleBaseTest {
+
+  @Test
+  public void ensureTestClassInstantiationWorksWhenFactoryMethodAndCustomConstructorPresent() {
+    TestNG testng = create(TestclassSample.class);
+    testng.setVerbose(2);
+    testng.run();
+    List<String> expected = Arrays.asList("1", "2");
+    assertThat(TestclassSample.logs).containsExactlyInAnyOrderElementsOf(expected);
+  }
+
+}

--- a/src/test/java/test/factory/issue1924/TestclassSample.java
+++ b/src/test/java/test/factory/issue1924/TestclassSample.java
@@ -1,0 +1,30 @@
+package test.factory.issue1924;
+
+import java.util.List;
+import org.testng.Reporter;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.testng.collections.Lists;
+
+public class TestclassSample {
+  public static List<String> logs = Lists.newArrayList();
+
+  private int i;
+
+  public TestclassSample(int i) {
+    this.i = i;
+  }
+
+  @Factory
+  public static Object[] produce() {
+    return new Object[]{
+        new TestclassSample(1),
+        new TestclassSample(2)
+    };
+  }
+
+  @Test
+  public void testMethod() {
+    logs.add(Integer.toString(this.i));
+  }
+}

--- a/src/test/java/test/priority/PriorityTest.java
+++ b/src/test/java/test/priority/PriorityTest.java
@@ -1,8 +1,7 @@
 package test.priority;
 
 import java.util.stream.Collectors;
-import org.assertj.core.api.Assertions;
-import org.testng.Assert;
+import static org.assertj.core.api.Assertions.assertThat;
 import org.testng.TestNG;
 import org.testng.annotations.Test;
 import org.testng.xml.XmlSuite;
@@ -23,7 +22,12 @@ public class PriorityTest extends SimpleBaseTest {
       tng.setParallel(XmlSuite.ParallelMode.METHODS);
     }
     tng.run();
-    Assert.assertEquals(listener.getInvokedMethodNames().toArray(), methods);
+    if (parallel) {
+      //If tests are being executed in parallel, then order of methods is non-deterministic.
+      assertThat(listener.getInvokedMethodNames()).containsExactlyInAnyOrder(methods);
+    } else {
+      assertThat(listener.getInvokedMethodNames()).containsExactly(methods);
+    }
   }
 
   @Test(enabled = false, description = "Make sure priorities work in parallel mode")
@@ -71,7 +75,7 @@ public class PriorityTest extends SimpleBaseTest {
         .stream()
         .filter(each -> !allSkipped.contains(each))
         .collect(Collectors.toList());
-    Assertions.assertThat(actual).containsExactlyElementsOf(expected);
+    assertThat(actual).containsExactlyElementsOf(expected);
   }
 
 }

--- a/src/test/resources/testng.xml
+++ b/src/test/resources/testng.xml
@@ -181,6 +181,7 @@
       <class name="test.github1461.MemoryLeakTestNg" />
       <!-- TODO fix the random issue <class name="test.issue565.Issue565Test"/>-->
       <class name="test.factory.issue1041.IssueTest"/>
+      <class name="test.factory.issue1924.IssueTest"/>
       <class name="test.factory.github328.GitHub328Test" />
       <class name="test.factory.issue1745.Github1745Test"/>
       <class name="test.objectfactory.github1827.GitHub1827Test"/>

